### PR TITLE
feat(ui5-writer): abstract functions

### PR DESCRIPTION
**Pull request by cianmSAP**
#1039

### CODE REVIEW

File: packages/project-access/src/file/file-search.ts [hunk 1]
```diff
@@ -136,3 +137,18 @@ export async function findFileUp(fileName: string, startPath: string, fs?: Edito
...
+ * recursing through all subdirectories
+ * @param {string} dir - the directory to walk
+ * @returns {string[]} - array of file path strings
+ * @throws if an error occurs reading a file path
+ */
+export function getFilePaths(dir: string): string[] | [] {
+    return readdirSync(dir).reduce((files: string[], entry: string) => {
+        const entryPath = join(dir, entry);
+        const isDirectory = statSync(entryPath).isDirectory();
+        return isDirectory ? [...files, ...getFilePaths(entryPath)] : [...files, entryPath];
```
Comment (Klaus-Keller) [inline packages/project-access/src/file/file-search.ts:152]: Hi @cianmSAP, can we please use asynchronous functions for file access to keep it in alignment with the rest of this module.

**Review by Klaus-Keller — CHANGES_REQUESTED** [review]

File: packages/project-access/src/file/file-search.ts (hunk 1 above)
Comment (cianmSAP) [reply packages/project-access/src/file/file-search.ts]: Updated. It resulted in a lot of updates needed in packages/ui5-application-writer/src/options.ts to await the result...

**Review by devinea — APPROVED** [review]
Thanks @cianmSAP 
Refactoring is as described. 
Ran UX_DEBUG tests locally and all looks good.

**Review by Klaus-Keller — APPROVED** [review]
- I focused review on `@sap-ux/project-access`, code changes look good
- Test coverage looks good
- Did not build and test locally
- The code smell regarding duplicate enum values is hard to address, because the two different keys point currently at the same value

Approved from my side

**Review by devinea — APPROVED** [review]

**Review by Klaus-Keller — APPROVED** [review]
No code changes since last review, just merge of main into this branch and pipeline restart.

Approved again.

**Review by tobiasqueck — APPROVED** [review]
Nothing to add to the other two approvals, no issues found in coding or changeset.
Just a comment (no need to change it though): the `async` `await` in the factories isn't necessary, just the change to `Promise<void>` in the factory declaration should have been necessary.

**Pull request merged**